### DISCUSSION TITLE
Restoring a backup to a new phone

### DISCUSSION THREAD

**Discussion by gittbuh** [seedvault-app/seedvault#331] [Q&A]
Hi,

Can a backup be restored to a phone that was not used to create the backup? Like for example, I buy a new phone or phone gets stolen and I want to recover my data easily. 

Thanks.

**Comment by nickcalyx** [comment]
Yes.  It should even work across different Android-based OS's.

**Comment by grote** [comment]
You'd need Seedvault on that new phone and it shouldn't be an older version of the OS and Seedvault.

**Reply by gittbuh** [reply]
I am testing this on a new user profile (I have previously logged into my nextcloud account where the backups are in the new profile) in my phone (with graphene os) and I am getting an "incorrect recovery code" message... Any ideas why? The code is not incorrect so it does not make sense.

Thanks in advance for the help.

**Reply by grote** [reply]
Can you provide a screenshot of the message you are getting and ideally logcat information?

**Reply by gittbuh** [reply]
Hi, thanks for the reply. Here is the screeshot. I checked the recovery code in the profile I used to create the backups and it works. So the recovery code is correct. I guess logcat is some sort of log? I do not know where to find it exactly and how to send it. Will there be any sensitive data there? 
[
![Screenshot_20211101-211617](https://user-images.githubusercontent.com/65078910/139743652-327fe6fe-911a-4c40-98ea-41aaa9bc7af2.png)
](url)

**Reply by grote** [reply]
One explanation for what is happening here could be that both profiles using their own Seedvault data created a different recovery code when set up the first time. Now you try the same code in both and it isn't matching. The screenshot indicates that you use the verify code screen, not the restore backup screen.

**Reply by gittbuh** [reply]
It is not that, I have obviously not set up the recovery code of the second profile. But in order to download the backup from the first profile into the second one, I have to use the recovery code from the first profile in the second one... or how am I supposed to procede otherwise?  Because if I simply try to go and "restore backup (experimental)", I get the error: "No suitable backups found at given location. This is most likely due to a wrong recovery code or a storage error". So I supposed I first had to "authenticate" myself with the recovery code from the first profile... Is it no so that it should work?

**Reply by grote** [reply]
> I have obviously not set up the recovery code of the second profile. But in order to download the backup from the first profile into the second one, I have to use the recovery code from the first profile in the second one...

Alright, but the screenshot you sent is from verifying an existing code not from entering a new one for restore.

How do you start the restore in the second profile? Ideally, you document the entire process with screenshots.

**Comment by noci2012** [comment]
Similar issue:
I have a GrapheneOS (13) on a pixel 4a, i want to migrate all files from there to a new one using GrapheneOS (14) on pixel 8.
On first attempt the passwords were asked, and entered. (and later verified to be sure).
On an incremental backup on USB stick it wasn't recognized as a valid backups...

I recreated the backup as a complete new backup on the pixel 4a, then the stick was recognized sort of....
In the backup menu the restore option stays greyed out.   selecting RESTORE through the keyboard
shows: An error occurred while loading the backups..

So how to proceed.

(I attempted to get a logcat ... there are hardly any seedvault messages in there. (Only when leaving the app it shows some).
The USB cable obviously needs to be removed to allow the USB stick for restore.)

**Reply by grote** [reply]
you could try capturing a log cat via wireless ADB. did you run fsck on the stick or try a different one? maybe it is faulty?

Restore is meant to be part of SetupWizard. Not sure how Graphene does that. Also not sure why restore option would be greyed out. After SUW, official Seedvault doesn't even have a restore option.

**Reply by grote** [reply]
> In the backup menu the restore option stays greyed out. selecting RESTORE through the keyboard
shows: An error occurred while loading the backups..

Note that this is not officially supported and enabled by Graphene with a patch. The official way to restore is during set-up of the phone  when setup wizard runs.
